File: datasets/C++/Mistral/Algorithm/easy1.cpp
#include <iostream>
#include <vector>
#include <unordered_set>
#include <algorithm>

using namespace std;

int alternate(const string &s) {
    unordered_set<char> chars(s.begin(), s.end());
    int max_length = 0;

    for (char char1 : chars) {
        for (char char2 : chars) {
            if (char1 == char2) continue;

            string filtered;
            for (char c : s) {
                if (c == char1 || c == char2) {
                    filtered += c;
                }
            }

            bool valid = true;
            for (size_t i = 0; i < filtered.size() - 1; ++i) {
                if (filtered[i] == filtered[i + 1]) {
                    valid = false;
                    break;
                }
            }

            if (valid && filtered.size() > max_length) {
                max_length = filtered.size();
            }
        }
    }

    return max_length;
}

int main() {
    int length;
    cout << "Enter the length of the string: ";
    cin >> length;
    string s;
    cout << "Enter the string: ";
    cin >> s;

    int result = alternate(s);
    cout << result << endl;

    return 0;
}

File: datasets/Python/Mistral/Algorithm/easy1.py
def alternate(s):
    # Generate all pairs of characters
    chars = set(s)
    max_length = 0

    # Check all possible pairs
    for char1 in chars:
        for char2 in chars:
            if char1 == char2:
                continue
            # Create a new string by removing all characters except char1 and char2
            filtered = [c for c in s if c == char1 or c == char2]
            valid = True
            # Check if the filtered string is alternating
            for i in range(len(filtered) - 1):
                if filtered[i] == filtered[i + 1]:
                    valid = False
                    break
            if valid and len(filtered) > max_length:
                max_length = len(filtered)

    return max_length

# Input reading
length = int(input("Enter the length of the string: "))
s = input("Enter the string: ")

# Function call
result = alternate(s)
print(result)